07-errors.md

```diff
@@ -0,0 +1,311 @@
+| 5     | `SysErrInvalidReceiver`    | receiver doesn't exist and can't be automatically created  |
+| 6     | `SysErrInsufficientFunds`  | sender didn't have the requisite funds                     |
+| 7     | `SysErrOutOfGas`           | out of gas                                                 |
+| 8     | reserved                   |                                                            |
+| 9     | `SysErrIllegalExitCode`    | message receiver aborted with an system exit code          |
+| 10    | `SysErrAssertionFailed`    | some internal assertion failed                             |
+| 11-15 | reserved                   |                                                            |
+
+Notes:
+
+1. If a message sender doesn't exist on-chain, the VM will return a receipt with the
+   `ExitCode::SysErrSenderInvalid` exit code.
+2. If an actor calls `vm::abort` with a system exit code, that exit code will be replaced with
+   `ExitCode::SysErrIllegalExitCode`.
+3. If an actor panics/crashes/etc., the exit code `ExitCode::SysErrIllegalInstruction` will be
+   returned.
+4. If a message runs out of gas, the VM will return a receipt with the `ExitCode::SysErrOutOfGas`
+   exit code.
+5. Most of these exit codes are only reflected on-chain and will not be observed in internal sends.
+   The exceptions are:
+    - `SysErrIllegalExitCode` if the called actor aborted with a system exit code.
+    - `SysErrIllegalInstruction` if the called actor paniced.
+
+Changes from pre-FVM Filecoin:
+
+1. Removed `ExitCode::SysErrInvalidMethod` (3). Dispatch is now handled inside actor code.
+2. Removed `ExitCode::SysErrForbidden` (8). Caller verification is a userspace operation, we'll now
+   exit with `ExitCode::ErrForbidden`.
+4. Replaced `ExitCode::SysErrIllegalArgument` with `ExitCode::SysErrAssertionFailed` to more
+   accurately reflect its usage. When this appears on-chain, it means that a message hit some
+   internal Filecoin assert that shouldn't happen.
+
+### Standard Non-Zero Exit  Codes
+
+Standard non-zero exit codes are codes returned by actors to indicate common failure conditions.
+
+| Code  | Status                 | Description                                         |
+|-------|------------------------|-----------------------------------------------------|
+| 16    | `ErrIllegalArgument`   | invalid message parameters                          |
+| 17    | `ErrNotFound`          | message referenced something that doesn't exist     |
+| 18    | `ErrForbidden`         | operation forbidden                                 |
+| 19    | `ErrInsufficientFunds` | insufficient funds for operation                    |
+| 20    | `ErrIllegalState`      | the actor is in an illegal state                    |
+| 21    | `ErrSerialization`     | actor failed to serialize or deserialize some state |
+| 22    | `ErrUnhandledMessage`  | actor cannot handle this message                    |
```

Comment:
This name is great as (a) it decouples from the method number as the dispatch field, (b) it decouples from _methods_ being the unit of invokable functionality. Hence it suits actor orientation much better.